WISE-Community/WISE
# Cache teacher homepage runs and projects

Added a `/teacher/home` parent route, which includes child `/teacher/home/schedule` and `/teacher/home/library/...` routes. When navigating to any `/teacher/home/...` route for the first time, the user's runs and library projects are loaded once and then stored until user navigates away from `/teacher/home`.

Also fixed automatic scrolling to new runs/projects.

To test:
- Log in as a teacher and ensure that teacher's runs and all library projects load when redirected to `/teacher/home/schedule`.
- Click on the "Browse WISE Units" tab. Library units should display with minimal delay. Ensure that no new server requests are made. Switch between library tabs and make sure units show up correctly.
- Try searching and filtering the Class Schedule and the library. Switch back and forth between schedule and different library tabs. Search and filter choices should persist.
- Copy a unit. After the copy completes, you should be switched to the first page of 'My Units' tab (if not already there). Ensure that the new unit flashes a few times and has a light yellow-orange background.
- Create a new run. After the run is created and you click 'Done' on the confirmation dialog, you should be taken to the Class Schedule. Make sure the page scrolls to the new run and the new run has a light yellow-orange background.
- Load the WISE homepage. Make sure the homepage library loads correctly (and new server request is made to retrieve the official library units). Make sure searching and filtering works.
- Log in as a student and add a run. Page should scroll to the new run. New run should appear with light yellow-orange background.

Closes #1830
